### PR TITLE
add users and carts

### PR DESCRIPTION
**Pull request by npdoan1996**

### CODE REVIEW

File: src/routes/users.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express'; 
+import models from '../models';
+
+const router = Router(); 
+
+// getCartByUserID()
+router.get('/:userId', (req, res) => {
```
Comment (itsme2980) [inline src/routes/users.js]: Add prefix name for api-path for clearer. Ref: https://github.com/LeReOrg/be/blob/ee303ee8572078344ead70cfcf46bb63064bf46f/src/routes/products.js#L21

File: src/routes/carts.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Router } from 'express'; 
+import models from '../models';
+
+const router = Router(); 
+
+// getProductsByCartID()
+router.get('/:cartId', (req, res) => {
```
Comment (itsme2980) [inline src/routes/carts.js]: Add prefix name for api-path for clearer. Ref: https://github.com/LeReOrg/be/blob/ee303ee8572078344ead70cfcf46bb63064bf46f/src/routes/products.js#L21

File: src/index.js [hunk 3]
```diff
@@ -11,63 +11,21 @@ const PORT = 3000;
...
 
-// // getCategory()
-// app.get('/categories', (req, res) => {
-//   res.send(models.categories);
-// });
-
 // getTopProduct()
 app.get('/topProducts', (req, res) => {
   res.send(models.products);
 });
```
Comment (itsme2980) [inline src/index.js]: Should we move it into `routes.products` ?

Comment (npdoan1996) [reply src/index.js]: Incomplete API. Change to getProducts

**Review by itsme2980 — APPROVED** [review]

File: src/routes/users.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express'; 
+import models from '../models';
+
+const router = Router(); 
+
+// getCartByUserID()
+router.get('/getCartByUserId/:userId', (req, res) => {
+    let cart = models.carts.filter(cart => cart.id === req.params["userId"]);
```
Comment (itsme2980) [inline src/routes/users.js]: `cart.id` should be `cart.owner_id`

File: src/routes/products.js [hunk 5]
```diff
@@ -1,7 +1,12 @@
 import { Router } from 'express';
-import models from './../models';
+import models from '../models';
 
 const router = Router();
+
+//getProducts
+router.get('/', (req, res) => {
```
Comment (itsme2980) [inline src/routes/products.js]: add prefix here

**Pull request merged**